//
// Bargaining Protocol messages
//
// Use fields 1000+ for extensions;
//

package bargaining;
option java_package = "org.bitcoin.protocols.bargaining";
option java_outer_classname = "Protos";


// Generalized form of "send payment to this/these bitcoin addresses"
message Output {
	optional uint64 amount = 1 [default = 0]; // amount is integer-number-of-satoshis
	required bytes script = 2; // usually one of the standard Script forms
}


// BargainingMessage
message BargainingMessage {
	required string msg_type = 1;
  	optional uint32 details_version = 2 [default = 1];
  	required bytes serialized_details = 3;
  	optional string sign_type = 4 [default = "none"];
  	optional bytes sign_data = 5;
  	optional bytes signature = 6;
}


// BargainingRequestDetails
message BargainingRequestDetails {
  	optional string network = 1 [default = "main"];
  	optional bytes buyer_data = 2;
	optional bytes seller_data = 3;
	required uint64 time = 4;
  	optional uint64 expires = 5;
  	optional string bargaining_url = 6;
}
 
 
// BargainingRequestACKDetails
message BargainingRequestACKDetails {
  	optional string network = 1 [default = "main"];
  	optional bytes buyer_data = 2;
  	optional bytes seller_data = 3;
  	required uint64 time = 4;
  	optional uint64 expires = 5;
  	optional string bargaining_url = 6;
  	repeated Output outputs = 7;
  	optional string memo = 8;
}


// BargainingProposalDetails
message BargainingProposalDetails {
  	optional bytes buyer_data = 1;
  	optional bytes seller_data = 2;
  	required uint64 time = 3;
  	repeated bytes transactions = 4;
  	repeated Output refund_to = 5;
  	optional string memo = 6;
}


// BargainingProposalACKDetails
message BargainingProposalACKDetails {
  	optional bytes buyer_data = 1;
  	optional bytes seller_data = 2;
  	required uint64 time = 3;
  	repeated Output outputs = 4;
  	optional string memo = 5;
}


// BargainingCompletionDetails
message BargainingCompletionDetails {
  	optional bytes buyer_data = 1;
  	optional bytes seller_data = 2;
  	required uint64 time = 3;
  	repeated bytes transactions = 4;
  	optional string memo = 5;
}


// BargainingCancellationDetails
message BargainingCancellationDetails {
  	optional bytes buyer_data = 1;
  	optional bytes seller_data = 2;
  	required uint64 time = 3;
  	optional string memo = 4;
}



